### DISCUSSION TITLE
No commands run in the initial microVM

### DISCUSSION THREAD

**Discussion by imyunha** [firecracker-microvm/firecracker#4239] [Q&A] [answered]
Hello, everyone. I'm a newbie of firecracker.

I'm following [this](https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md).
After I run `ssh -i ./ubuntu-22.04.id_rsa root@172.16.0.2` , ping, apt, ... all commands don't work.
![image](https://github.com/firecracker-microvm/firecracker/assets/61413047/ebae47ae-6e64-4524-8b60-fc44f5d61e87)

I added `nameserver 8.8.8.8` to `/etc/resolv.conf`, but it doesn't work. Any other solutions?

**Comment by wearyzen** [comment, marked as answer]
Hi @choiyhking, getting started guide just covers how to quickly bootup a uVM and test it locally. To connect the uVM to the internet you'll have to follow some more steps from https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md.

Sharing reference steps to follow along with the getting started page:
1. Before starting Firecracker, run the [On The Host](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md#on-the-host) steps.
2. In the steps from Getting started guide, make sure to use "eth0" instead of "net1" (shared command below for reference)
> curl -X PUT --unix-socket "${API_SOCKET}" \
    --data "{
        \"iface_id\": \"eth0\",
        \"guest_mac\": \"$FC_MAC\",
        \"host_dev_name\": \"$TAP_DEV\"
    }" \
    "http://localhost/network-interfaces/eth0"
3. After the uVM boots up run the [In The Guest](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md#in-the-guest) steps along with `echo nameserver 8.8.8.8 > /etc/resolv.conf`

Let me know if this helps.

**Reply by imyunha** [reply]
Hi @sudanl0, Thank you so much for your detail reply.
Originally guest to host ping didn't work either, but this now works fine.
But `ping google.com`, `apt update` still doesn't work. 

Should I proceed with the [advanced part](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md#advanced-setting-up-a-bridge-interface)?
(Actually, I'm not good at linux network..it's more difficult to solve)

For now, I'm connected to wi-fi. Is this can be a problem?

**Reply by wearyzen** [reply]
Hi @choiyhking,
The current steps should have worked to get the ping started.
The networking guide assumes that eth0 is the internet-facing interface but since you use wifi you might need to modify the `On Host` steps to replace eth0 with the name of your wifi interface as below (it probably would be wlan0):
```
sudo iptables -t nat -A POSTROUTING -o `YOUR_WIFI_INTERFACE_NAME` -j MASQUERADE
sudo iptables -A FORWARD -i tap0 -o `YOUR_WIFI_INTERFACE_NAME` -j ACCEPT
```

Could you confirm if using the right wifi interface name on the host works for you?

**Reply by imyunha** [reply]
After I connected my device to ethernet, I followed [getting-started](https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md#starting-firecracker) and [network-setup](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md).

But in [getting-started](https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md#starting-firecracker), I modified `MASK_SHORT="/30"` to `MASK_SHORT="/24"` and used `eth0` instead of `net1`.
And in [network-setup](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md), I only used [in the guest part](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md#:~:text=your%20command%20line.-,In%20The%20Guest,-Once%20you%20have).

As a result, network problem is solved. 

But, `apt update` made `E: You don't have enough free space in /var/cache/apt/archives/.`

![image](https://github.com/firecracker-microvm/firecracker/assets/61413047/d166157e-eff3-4c99-b535-735fccf589b4)
It looks like rootfs size is too small...how can I size-up?
Should I reference [this](https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md)?

I really appreciate to @sudanl0.😊😊

**Reply by wearyzen** [reply]
The rootfs mentioned [here](https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md#getting-a-rootfs-and-guest-kernel-image) should already be up to date but yes, in general if you want to try a custom rootfs the link mentioned is the right one.

**Reply by imyunha** [reply]
Thank you so much. I'll try it. :)
